minio/api.py

```diff
@@ -298,8 +298,11 @@ def list_buckets(self):
 
         if response.status != 200:
             raise ResponseError(response, method).get_exception()
-
-        return parse_list_buckets(response.data)
+        try:
+            return parse_list_buckets(response.data)
+        except InvalidXMLError:
+            if self._endpoint_url.endswith("s3.amazonaws.com"):
+                raise InvalidArgumentError('Invalid access_key and secret_key.')
```

Comment:
I think `Invalid access key or secret ley` is not appropriate here.  How about return S3 error `ACCESS_DENIED` or `METHOD_NOT_ALLOWED` ?